### PR TITLE
Moved api responses to response handler

### PR DESCRIPTION
**Pull request by aolundin**
- renamed responseParser to responseHandler
- Moved response logic from routes to responseHandler

### CODE REVIEW

File: app/middleware/authentication.middleware.js [hunk 1]
```diff
@@ -3,26 +3,23 @@
...
-                userController.createNewUser(name, email)
-                    .then(next);
-            } else {
-                next();
-            }
-        })
-        .catch(response.send);
+    
+    userController.createUserIfNonexistent(name, email)
+        .then(responseHandler.sendNext(next))
```
Comment (Tobbe) [inline app/middleware/authentication.middleware.js]: Perhaps `sendToNext` is more clear?

**Comment by aolundin** [comment]
agreed. it's been changed

File: app/routes/attribute.routes.js [hunk 2]
```diff
@@ -5,30 +5,22 @@
...
     // create an attribute
     routes.post('/', function(request, response) {
         attributeController.createNewAttribute(request.body)
-            .then(function(attribute) {
-                return response.json(attribute);
-            })
-            .catch(function(error) {
-                return response.status(error.status || 500).send(error.message);
-            });
+            .then(response.json.bind(response))
```
Comment (Tobbe) [inline app/routes/attribute.routes.js]: I agree that the methods become short and sweet like this, but I wonder if it wouldn't be more readable to create a small private helper function called for example `sendJsonResponse` and pass that to `then`. That way it'd read something like "create new attribute, then send JSON response". What do you think?

**Comment by aolundin** [comment]
added function sendJsonResponse in responseHandler

File: app/utils/response.handler.js [hunk 3]
```diff
@@ -73,4 +73,23 @@ exports.parseGet = function(response) {
...
+exports.sendToNext = function(next) {
+    return function() {
+        next();
+    };
+};
+
+exports.sendJsonResponse = function(response) {
+    return function(object) {
+        return response.json(object);
+        //return response.json.bind(response))
```
Comment (Tobbe) [inline app/utils/response.handler.js]: remove

**Comment by aolundin** [comment]
removed comment

File: app/utils/response.handler.js [hunk 4]
```diff
@@ -73,4 +73,22 @@ exports.parseGet = function(response) {
...
+exports.sendToNext = function(next) {
+    return function() {
+        next();
+    };
+};
+
+exports.sendJsonResponse = function(response) {
+    return function(object) {
+        return response.json(object);
+        }
```
Comment (Tobbe) [inline app/utils/response.handler.js]: indentation...

**Pull request merged**